lib/score.js

```diff
@@ -0,0 +1,40 @@
+const Rules = require('../lib/rules.js');
+
+class Score {
+  constructor(scorecard, RulesClass = Rules) {
+    this.framesArray = scorecard.framesArray;
+    this.rulesClassInstance = new RulesClass();
+  }
+
+  calculateScore = () => {
+    let [score, index] = [0, 0];
+    const flattenedFramesArray = this.framesArray.flat();
```

Comment:
Flattening an array is very clever. Didn't know there was a flat built-in function.